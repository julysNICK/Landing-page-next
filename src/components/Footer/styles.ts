import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../Textcomponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
