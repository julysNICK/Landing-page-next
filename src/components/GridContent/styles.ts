import styled, { css } from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
