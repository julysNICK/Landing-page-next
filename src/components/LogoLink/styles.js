import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
    > img {
      height: 3rem;
    }
  `}
`;
