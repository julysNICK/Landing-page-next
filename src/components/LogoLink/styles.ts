import styled from 'styled-components';

export const Container = styled.a`
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
  > img {
    height: 3rem;
  }
`;
