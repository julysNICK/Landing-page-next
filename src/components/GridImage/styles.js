import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
