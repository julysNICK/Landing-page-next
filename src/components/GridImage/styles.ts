import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
