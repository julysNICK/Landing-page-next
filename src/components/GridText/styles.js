import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
