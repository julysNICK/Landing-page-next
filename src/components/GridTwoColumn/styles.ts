import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.font.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img`
  width: 100%;
`;
