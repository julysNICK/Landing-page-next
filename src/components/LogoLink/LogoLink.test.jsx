import { screen } from '@testing-library/dom';
import { LogoLink } from '.';
import { renderTheme } from '../../styles/render-theme';
describe('<LogoLink />', () => {
  it('should  render text logo', () => {
    renderTheme(<LogoLink link="#target" text="olá mundo" />);
    expect(
      screen.getByRole('heading', { name: 'olá mundo' }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'olá mundo' })).toHaveAttribute(
      'href',
      '#target',
    );
  });

  it('should  render text logo', () => {
    renderTheme(
      <LogoLink link="#target" text="olá mundo" srcImg="image.jpg" />,
    );

    expect(screen.getByAltText('olá mundo')).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });
  it('should  render text logo', () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="olá mundo" srcImg="image.jpg" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
