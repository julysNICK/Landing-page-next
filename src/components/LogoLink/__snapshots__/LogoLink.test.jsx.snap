// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should  render text logo 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 > img {
  height: 3rem;
}

.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="olÃ¡ mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
